File: core/src/main/java/org/ecos/reader/core/io/files/ZipNavigatorServiceImpl.java
package org.ecos.reader.core.io.files;

import org.ecos.reader.core.io.files.exceptions.DoNoExistsException;

import reader.ecos.org.core.io.files.exceptions.UnknownError;

@SuppressWarnings("unused")
public class ZipNavigatorServiceImpl implements ZipNavigatorService {
    @Override
    public FileCollection getFilesInside(ZipFile file) throws DoNoExistsException, UnknownError {

        if(!file.exists())
            throw new DoNoExistsException("");

        return file.getFiles();
    }

}

File: core/src/main/kotlin/org/ecos/reader/core/io/files/ZipNavigatorServiceImpl.kt
package org.ecos.reader.core.io.files

import org.ecos.reader.core.io.files.exceptions.DoNoExistsException

import org.ecos.reader.core.io.files.exceptions.UnknownError

@Suppress("unused")
class ZipNavigatorServiceImpl : ZipNavigatorService {
    @Throws(DoNoExistsException::class, UnknownError::class)
    override fun getFilesInside(file: ZipFile): FileCollection {
        if (!file.exists())
            throw DoNoExistsException("")

        return file.files
    }
}